monkey/monkey_island/cc/server_setup.py

```diff
@@ -60,6 +92,8 @@
         logger.warning("Setup only flag passed. Exiting.")
         return
 
+    bootloader_server_thread = _start_bootloader_server(MONGO_URL)
```

Comment:
Bootloader server is a separate server and doesn't care about island's initialization workflow or parameters. I don't think this belongs in `_start_island_server`, since it has nothing to do with mongodb, env_singleton or anything else in this method.